testdata/vote.txt

```diff
@@ -0,0 +1,97 @@
+# Tests that a node that is behind on the log will lose the vote.
```

Comment:
I think it would be better to write this test in the first commit in such a way that it demonstrates the problem. As far as I understand, the problem is that a node that failed to be elected, can try campaigning again and lose again. Can we recreate this behaviour before fixing it?

It might be necessary to place this commit after the newly added `datadriven` commands/flags, which would help reproing this issue.

Reply:
> As far as I understand, the problem is that a node that failed to be elected, can try campaigning again and lose again. Can we recreate this behaviour before fixing it?


We can accurately control the behaviour using the datadriven test. No matter how many times the node 2 retries the campaign, it will always lose the campaign as long as it's lag behind. Actually I thought about letting node 2 retry campaign after it lost the vote, but it doesn't make much sense. There isn't difference between trying 1 time and try N (>=2) times. It also complicates the test case, at least much longer.

As I mentioned previously,  Ideally we should have a tool to measure the average recovery time, something like what #86 does. Probably we should discuss/think about it separately.


> It might be necessary to place this commit after the newly added `datadriven` commands/flags, which would help reproing this issue.

The new flags & command is only for updating the test case (for the 5th commit). Actually we can even add the new flags & command in a separate PR if you insist, but it seems not a big deal.

Reply:
I think we could still make this test in such a way that before/after behaviour is more obvious. See the comment below.